detectors/aws/ecs/ecs.go

```diff
@@ -43,27 +43,32 @@ var (
 )
 
 // Create interface for methods needing to be mocked
-type detectorUtils interface {
+type detectorUtilsResources interface {
 	getContainerName() (string, error)
 	getContainerID() (string, error)
 }
 
-// struct implements detectorUtils interface
-type ecsDetectorUtils struct{}
+// struct implements detectorUtilsResources interface
+type DetectorUtils struct{}
 
 // resource detector collects resource information from Elastic Container Service environment
-type ResourceDetector struct {
-	utils detectorUtils
+type resourceDetector struct {
+	utils detectorUtilsResources
 }
 
-// compile time assertion that ecsDetectorUtils implements detectorUtils interface
-var _ detectorUtils = (*ecsDetectorUtils)(nil)
+// compile time assertion that ecsDetectorUtils implements detectorUtilsResources interface
+var _ detectorUtilsResources = (*DetectorUtils)(nil)
 
 // compile time assertion that resource detector implements the resource.Detector interface.
-var _ resource.Detector = (*ResourceDetector)(nil)
+var _ resource.Detector = (*resourceDetector)(nil)
+
+// returns resource detector struct
+func NewResourceDetector(detectorUtils detectorUtilsResources) resource.Detector {
```

Comment:
The detectorUtils of the resource interface is an abstraction introduced for testing purposes. We should not leak this implementation detail by requiring a value be passed here.

```suggestion
func NewResourceDetector() resource.Detector {
```

The testing for this detector, since it has internal access to the private fields and types of this package, can assert and then override the needed fields for testing still.